servers/quarkus-server/src/main/java/org/projectnessie/server/grpc/TreeService.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.server.grpc;
+
+import static org.projectnessie.grpc.ProtoUtil.fromProto;
+import static org.projectnessie.grpc.ProtoUtil.refFromProto;
+import static org.projectnessie.grpc.ProtoUtil.refToProto;
+import static org.projectnessie.grpc.ProtoUtil.toProto;
+
+import io.grpc.stub.StreamObserver;
+import io.quarkus.grpc.GrpcService;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import org.projectnessie.api.grpc.AssignReferenceRequest;
+import org.projectnessie.api.grpc.CommitLogRequest;
+import org.projectnessie.api.grpc.CommitLogResponse;
+import org.projectnessie.api.grpc.CommitRequest;
+import org.projectnessie.api.grpc.DeleteReferenceRequest;
+import org.projectnessie.api.grpc.Empty;
+import org.projectnessie.api.grpc.EntriesRequest;
+import org.projectnessie.api.grpc.EntriesResponse;
+import org.projectnessie.api.grpc.GetAllReferencesResponse;
+import org.projectnessie.api.grpc.GetReferenceByNameRequest;
+import org.projectnessie.api.grpc.MergeRequest;
+import org.projectnessie.api.grpc.Reference;
+import org.projectnessie.api.grpc.TransplantRequest;
+import org.projectnessie.api.grpc.TreeServiceGrpc;
+import org.projectnessie.client.grpc.GrpcExceptionMapper;
+import org.projectnessie.grpc.ProtoUtil;
+import org.projectnessie.model.Branch;
+import org.projectnessie.model.ImmutableMerge;
+import org.projectnessie.model.ImmutableTransplant;
+import org.projectnessie.model.Tag;
+import org.projectnessie.services.rest.TreeResource;
+
+/** The gRPC service implementation for the Tree-API. */
+@GrpcService
+public class TreeService extends TreeServiceGrpc.TreeServiceImplBase {
+
+  private final TreeResource treeResource;
+
+  @Inject
+  public TreeService(TreeResource treeResource) {
+    this.treeResource = treeResource;
+  }
+
+  @Override
+  public void getAllReferences(Empty request, StreamObserver<GetAllReferencesResponse> observer) {
+    try {
+      observer.onNext(
+          GetAllReferencesResponse.newBuilder()
+              .addAllReference(
+                  treeResource.getAllReferences().stream()
+                      .map(ProtoUtil::refToProto)
+                      .collect(Collectors.toList()))
+              .build());
+      observer.onCompleted();
```

Comment:
The `onNext` + `onCompleted`/`onError` can probably be shortened / wrapped as you did on the client side

Reply:
done